File: app/src/main/java/me/imzack/app/end/injector/component/HomeComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.HomePresenterModule;
import me.imzack.app.end.injector.scope.ActivityScope;
import me.imzack.app.end.view.activity.HomeActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = HomePresenterModule.class)
public interface HomeComponent {

    void inject(HomeActivity homeActivity);
}

File: app/src/main/kotlin/me/imzack/app/end/injector/component/HomeComponent.kt
package me.imzack.app.end.injector.component

import dagger.Component
import me.imzack.app.end.injector.module.HomePresenterModule
import me.imzack.app.end.injector.scope.ActivityScope
import me.imzack.app.end.view.activity.HomeActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(HomePresenterModule::class))
interface HomeComponent {

    fun inject(homeActivity: HomeActivity)
}
